akka-stream/src/main/scala/akka/stream/SystemMaterializer.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import akka.actor.ActorSystem
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+
+/**
+ * The system materializer is a default materializer to use for most cases running streams, it is a single instance
+ * per actor system that is tied to the lifecycle of that system.
+ *
+ * Not intended to be manually used in user code.
+ */
+object SystemMaterializer extends ExtensionId[SystemMaterializer] with ExtensionIdProvider {
+  override def get(system: ActorSystem): SystemMaterializer = super.get(system)
+
+  override def lookup = SystemMaterializer
+
+  override def createExtension(system: ExtendedActorSystem): SystemMaterializer =
+    new SystemMaterializer(system)
+}
+
+final class SystemMaterializer(_system: ExtendedActorSystem) extends Extension {
+  val materializer = {
+    val settings = ActorMaterializerSettings(_system)
+    ActorMaterializer.systemMaterializer(settings, "default", _system)
+  }
+  // FIXME not 100% sure about this, it will anyways stop when the system stops the actors
+  _system.registerOnTermination {
+    materializer.shutdown()
```

Comment:
That will anyway run after stopping actors so probably no point in doing it. If `materializer.shutdown` is more graceful than stopping the actor we should add it as a CoordinatedShutdown task.

Reply:
I don't think it is more graceful, I'll remove this.